Sources/web3swift/API/APIMethod.swift

```diff
@@ -0,0 +1,288 @@
+
+        case .getBalance(let address, let blockNumber):
+            return [RequestParameter.string(address), RequestParameter.string(blockNumber.stringValue)]
+
+        case .getStorageAt(let address, let hash, let blockNumber):
+            return [RequestParameter.string(address), RequestParameter.string(hash), RequestParameter.string(blockNumber.stringValue)]
+
+        case .getCode(let address, let blockNumber):
+            return [RequestParameter.string(address), RequestParameter.string(blockNumber.stringValue)]
+
+        case .getBlockByHash(let hash, let bool):
+            return [RequestParameter.string(hash), RequestParameter.bool(bool)]
+
+        case .getBlockByNumber(let block, let bool):
+            return [RequestParameter.string(block.stringValue), RequestParameter.bool(bool)]
+
+        case .feeHistory(let uInt, let blockNumber, let array):
+            return [RequestParameter.string(uInt.hexString), RequestParameter.string(blockNumber.stringValue), RequestParameter.doubleArray(array)]
+
+        case .createAccount(let string):
+            return [RequestParameter.string(string)]
+
+        case .unlockAccount(let address, let string, let uInt):
+            return [RequestParameter.string(address), RequestParameter.string(string), RequestParameter.uint(uInt ?? 0)]
+        }
+    }
+
+    var call: String {
+        switch self {
+        case .gasPrice: return "eth_gasPrice"
+        case .blockNumber: return "eth_blockNumber"
+        case .getNetwork: return "net_version"
+        case .getAccounts: return "eth_accounts"
+        case .sendRawTransaction: return "eth_sendRawTransaction"
+        case .sendTransaction: return "eth_sendTransaction"
+        case .getTransactionByHash: return "eth_getTransactionByHash"
+        case .getTransactionReceipt: return "eth_getTransactionReceipt"
+        case .personalSign: return "eth_sign"
+        case .getLogs: return "eth_getLogs"
+        case .call: return "eth_call"
+        case .estimateGas: return "eth_estimateGas"
+        case .getTransactionCount: return "eth_getTransactionCount"
+        case .getBalance: return "eth_getBalance"
+        case .getStorageAt: return "eth_getStorageAt"
+        case .getCode: return "eth_getCode"
+        case .getBlockByHash: return "eth_getBlockByHash"
+        case .getBlockByNumber: return "eth_getBlockByNumber"
+        case .feeHistory: return "eth_feeHistory"
+
+        case .unlockAccount: return "personal_unlockAccount"
+        case .createAccount: return "personal_createAccount"
+        case .getTxPoolStatus: return "txpool_status"
+        case .getTxPoolContent: return "txpool_content"
+        case .getTxPoolInspect: return "txpool_inspect"
+        }
+    }
+}
+
+extension APIRequest {
+    public static func sendRequest<Result>(with provider: Web3Provider, for call: APIRequest) async throws -> APIResponse<Result> {
```

Comment:
I doubt it.

I've considered to move this method to `web3httpprovider` class, but it's complicated because then you've should to make the whole provider struct generic and then make protocol that it conforms with associated type. Which leads to you need to rewrite almost every method in whole lib, so i refused it.

Making this method non static within this enum makes API kinda wired. I'll be looks like follow:

```swift
let request: APIRequest = .gasPrice
let response: APIResponse<BigUInt> = request.send(with: web3.provider)
```

So i've refused that either.